item.go

```diff
@@ -30,25 +30,34 @@ type Item[K comparable, V any] struct {
 	// well, so locking this mutex would be redundant.
 	// In other words, this mutex is only useful when these fields
 	// are being read from the outside (e.g. in event functions).
-	mu         sync.RWMutex
-	key        K
-	value      V
-	ttl        time.Duration
-	expiresAt  time.Time
-	queueIndex int
-	version    int64
+	mu            sync.RWMutex
+	key           K
+	value         V
+	ttl           time.Duration
+	expiresAt     time.Time
+	queueIndex    int
+	version       int64
+	calculateCost CostFunc[K, V]
+	cost          uint64
 }
 
 // NewItem creates a new cache item.
+// Deprecated, use NewItemWithOpts instead
 func NewItem[K comparable, V any](key K, value V, ttl time.Duration, enableVersionTracking bool) *Item[K, V] {
+	return NewItemWithOpts(key, value, ttl, WithVersionTracking[K, V](enableVersionTracking))
+}
+
+// NewItemWithOpts creates a new cache item.
+func NewItemWithOpts[K comparable, V any](key K, value V, ttl time.Duration, opts ...ItemOption[K, V]) *Item[K, V] {
 	item := &Item[K, V]{
-		key:   key,
-		value: value,
-		ttl:   ttl,
+		key:           key,
+		value:         value,
+		ttl:           ttl,
+		calculateCost: func(item *Item[K, V]) uint64 { return 0 },
 	}
 
-	if !enableVersionTracking {
-		item.version = -1
```

Comment:
The default version value (-1) should still be set. The version option might not be provided at all.

Reply:
But `-1` is not the default value for the `version`. It is (and also was before this PR) only set to `-1` if version tracking is disabled. This is now covered by the new `withVersionTracking` option, which is always used by the updated implementation of the `NewItem` function.

Reply:
done